src/codegen/encoding/mod.rs

```diff
@@ -0,0 +1,580 @@
+mod borsh_encoding;
+
+use crate::ast::{Namespace, RetrieveType, Type};
+use crate::codegen::cfg::{ControlFlowGraph, Instr};
+use crate::codegen::encoding::borsh_encoding::BorshEncoding;
+use crate::codegen::expression::load_storage;
+use crate::codegen::vartable::Vartable;
+use crate::codegen::{Builtin, Expression};
+use crate::Target;
+use num_bigint::BigInt;
+use solang_parser::pt::Loc;
+
+/// This trait should be implemented by all encoding methods (ethabi, Scale and Borsh), so that
+/// we have the same interface for creating encode and decode functions.
+pub(super) trait Encoding {
```

Comment:
There are a ton of traits in rust called `Encoding`.  We should try to avoid collisions. How about `RuntimeEncoding`?

Reply:
If we encode all arguments during compile time, it is not runtime encoding any more. What do you think of `ChainEncoding`?

Reply:
We don't use the word `Chain` in our source tree, best not to introduce new nomenclature. How about `AbiEncoding` ?

Reply:
`AbiEncoding` could be another option.